homeassistant/components/template/event.py

```diff
@@ -0,0 +1,238 @@
+
+DEVICE_CLASS_SCHEMA: Final = vol.All(vol.Lower, vol.Coerce(EventDeviceClass))
+
+EVENT_COMMON_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASS_SCHEMA,
+        vol.Required(CONF_EVENT_TYPE): cv.template,
+        vol.Required(CONF_EVENT_TYPES): cv.template,
+    }
+)
+
+EVENT_YAML_SCHEMA = EVENT_COMMON_SCHEMA.extend(
+    make_template_entity_common_modern_attributes_schema(DEFAULT_NAME).schema
+)
+
+
+EVENT_CONFIG_ENTRY_SCHEMA = EVENT_COMMON_SCHEMA.extend(
+    TEMPLATE_ENTITY_COMMON_CONFIG_ENTRY_SCHEMA.schema
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the template event."""
+    await async_setup_template_platform(
+        hass,
+        EVENT_DOMAIN,
+        config,
+        StateEventEntity,
+        TriggerEventEntity,
+        async_add_entities,
+        discovery_info,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Initialize config entry."""
+    await async_setup_template_entry(
+        hass,
+        config_entry,
+        async_add_entities,
+        StateEventEntity,
+        EVENT_CONFIG_ENTRY_SCHEMA,
+        True,
+    )
+
+
+@callback
+def async_create_preview_event(
+    hass: HomeAssistant, name: str, config: dict[str, Any]
+) -> StateEventEntity:
+    """Create a preview event."""
+    return async_setup_template_preview(
+        hass,
+        name,
+        config,
+        StateEventEntity,
+        EVENT_CONFIG_ENTRY_SCHEMA,
+    )
+
+
+class AbstractTemplateEvent(AbstractTemplateEntity, EventEntity):
+    """Representation of a template event features."""
+
+    _entity_id_format = ENTITY_ID_FORMAT
+
+    # The super init is not called because TemplateEntity and TriggerEntity will call AbstractTemplateEntity.__init__.
+    # This ensures that the __init__ on AbstractTemplateEntity is not called twice.
+    def __init__(self, config: dict[str, Any]) -> None:  # pylint: disable=super-init-not-called
+        """Initialize the features."""
+        self._event_type_template = config[CONF_EVENT_TYPE]
+        self._event_types_template = config[CONF_EVENT_TYPES]
+
+        self._attr_device_class = config.get(CONF_DEVICE_CLASS)
+
+        self._event_type = None
+        self._attr_event_types = []
+
+    @callback
+    def _update_event_types(self, event_types: Any) -> None:
+        """Update the event types from the template."""
+        if event_types in (None, "None", ""):
+            self._attr_event_types = []
+            return
+
+        if not isinstance(event_types, list):
+            _LOGGER.error(
+                (
+                    "Received invalid event_types list: %s for entity %s. Expected list of"
+                    " strings"
+                ),
```

Comment:
We just check for a list, do we need to also check that each element is a string?